src/main/java/org/apache/mesos/scheduler/plan/DefaultPlanManager.java

```diff
@@ -61,10 +61,34 @@ public Plan getPlan() {
     }
 
     @Override
-    public Optional<Block> getCurrentBlock() {
+    public Optional<Block> getCurrentBlock(List<Block> dirtiedAssets) {
```

Comment:
I think you're right.  I think moving to `Collection<String>` makes sense.  It's the most explicit regarding actual behavior.